### PR TITLE
JS Tic-Tac-Toe: Improve instructions order

### LINKED ISSUE #26280: JS Tic-Tac-Toe: Improve instructions/resource order
### Describe your suggestion

From a recent [Discord discussion](https://discord.com/channels/505093832157691914/505093832157691916/1148294848303071353), I felt the project instructions could be improved to more closely follow the organisation/workflow concepts presented in the [building a house from the inside out](https://www.ayweb.dev/blog/building-a-house-from-the-inside-out) resource linked in step 4.2.

Currently, the instructions allude towards making different objects/modules for different parts of the logic but the flow of the instructions feels like it can lead to trying to handle the game logic (players/board/gameflow) *and* display logic/DOM interaction at the same time - something I feel (like the build a house resource suggests) would be better off focusing on game logic first and only then do you introduce display logic.
I feel this order can reinforce the concept and benefits of separating logic into different objects and modules in a much clearer way than is currently being done.

A draft proposal for new instructions:

> 1. Set up your project with HTML, CSS and Javascript files and get the Git repo all set up.
> 2. You’re going to store the gameboard as an array inside of a Gameboard object, so start there! Your players are also going to be stored in objects, and you’re probably going to want an object to control the flow of the game itself.
>     1. Your main goal here is to have as little global code as possible. Try tucking everything away inside of a module or factory. Rule of thumb: if you only ever need ONE of something (gameBoard, displayController), wrap their factory in an IIFE. If you need multiples of something (players!), you needn't do this (so you can reuse the factory).
>     2. In this project, think carefully about where each bit of logic should reside. Each little piece of functionality should be able to fit in the game, player or gameboard objects. Take care to put them in “logical” places. Spending a little time brainstorming here can make your life much easier later!
>     3. If you’re having trouble, [Building a house from the inside out](https://www.ayweb.dev/blog/building-a-house-from-the-inside-out) is a great article that lays out a highly applicable example of how you might organize your code for this project.
> 3. If you read the resource from the previous step, you'll probably want to focus on getting a working game in the console first. Make sure you include logic that checks for when the game is over! You should be checking for all 3-in-a-rows and ties. Try to avoid thinking about the DOM and your HTML/CSS until your game is working first!
> 4. Once you have a working console game, create an object that will handle the display/DOM logic. Write a function that will render the contents of the gameboard array to the webpage (for now, you can always just fill the gameboard array with `"X"`s and`"O"`s for now just to see what's going on).
> 5. Write the functions that allow players to add marks to a specific spot on the board by interacting with the appropriate DOM elements (e.g. letting players click on a board square to place their marker). Don't forget the logic that keeps players from playing in spots that are already taken!
> 6. Clean up the interface to allow players to put in their names, include a button to start/restart the game and add a display element that shows the results upon game end!

I have also omitted the extra credit section following the discussion in #26172

Thoughts?

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-tic-tac-toe and https://www.theodinproject.com/paths/full-stack-javascript/courses/javascript

### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### (Optional) Discord Name

mao_sz

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
As per #26280, this is a proposal for revamped JS TTT instructions to address the following:
- Bringing the "Building a house" resource to an earlier step as it can provide great insight to both the overall approach as well as overall structure - more useful earlier on in the process than previously, where people may well have written plenty of code and already settled on a perhaps less ideal structure.
- Rewriting instructions for what things to write/focus on to be more consistent with the "building a house" resource approach (don't touch display stuff until game logic works in console first). This is to reinforce the idea that logic such as the display/DOM is not part of the game flow logic and how decoupling it (both in a module and also in development flow) can aid with more modular code and workflow.
- Reworded the suggestions about using an IIFE for singletons. I felt the original wording could be interpreted as "IIFEs are used in order to instantiate singletons" (I remember having this confusion at first) as opposed to dealing utilising how it alters scope. I remember not understanding how ES6 modules "meant IIFEs weren't as useful anymore" because my understanding was ES6 modules were to do with scope and IIFEs were used to make singletons.
- Removed optional step 7 (smart AI) as per #26172 

## This PR
- Rewrote instructions to address the above issues

## Issue
Closes #26280

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
*Gentle bump if that's alright* ^^'

**Comment by CouchofTomato** [comment]
@01zulfi 

Are you still handling this one mate?

**Review by 01zulfi — CHANGES_REQUESTED** [review]

File: javascript/organizing_your_javascript_code/project_tic_tac_toe.md [hunk 1]
```diff
@@ -5,16 +5,12 @@ We're making a Tic Tac Toe game you can play in your browser!
...
-   2. If you're having trouble, [Building a house from the inside out](https://www.ayweb.dev/blog/building-a-house-from-the-inside-out) is a great article that lays out a highly applicable example of how you might organize your code for this project.
-5. Build the logic that checks for when the game is over!  Should check for 3-in-a-row and a tie.
-6. Clean up the interface to allow players to put in their names, include a button to start/restart the game and add a display element that congratulates the winning player!
-7. Optional - If you're feeling ambitious create an AI so that a player can play against the computer!
-   1. Start by just getting the computer to make a random legal move.
-   2. Once you've gotten that, work on making the computer smart.  It is possible to create an unbeatable AI using the minimax algorithm (read about it [here](https://en.wikipedia.org/wiki/Minimax), some googling will help you out with this one)
-   3. If you get this running _definitely_ come show it off in the chatroom.  It's quite an accomplishment!
```
[the remark this answers is not in the document]
Comment (01zulfi) [reply javascript/organizing_your_javascript_code/project_tic_tac_toe.md:19]: Let's keep this change as per https://github.com/TheOdinProject/curriculum/issues/26172#issuecomment-1793410258

**Review by 01zulfi — APPROVED** [review]
lgtm :rocket:

**Pull request merged**